### PR TITLE
Fix some LGTM alerts

### PR DESCRIPTION
**Pull request by 1m2c3t4**
## Description
LGTM has raised a number of alerts from analysis of apache/cloudstack (https://lgtm.com/projects/g/apache/cloudstack/alerts/?mode=tree).
This pull request is to fix some of the problems raised. Specifically,
- 2 `Array index out of bounds` errors in NetworkOrchestrator
- 1 `Equals method does not inspect argument type` in DeployDestination
- 1 `Equals on incomparable types` in ApiResponseHelper
- 28 `Contradictory type checks` in ApiServer, HypervDirectConnectResource and Ovm3HypervisorResource

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

### CODE REVIEW

**Comment by GabrielBrascher** [comment]
@blueorangutan package

File: api/src/main/java/com/cloud/deploy/DeployDestination.java [hunk 1]
```diff
@@ -78,6 +78,10 @@ public int hashCode() {
 
     @Override
     public boolean equals(Object obj) {
+        if (obj == null)
+            return false;
+        if (!(obj instanceof DeployDestination))
+            return false;
         DeployDestination that = (DeployDestination)obj;
         if (_dc == null || that._dc == null) {
```
Comment (rafaelweingartner) [inline api/src/main/java/com/cloud/deploy/DeployDestination.java]: Besides what @GabrielBrascher suggested, I have one suggestion though.
Can you append the word `this.` to the `_dc` variable? Then, we would have `this._dc == null || that._dc == null`.

**Comment by GabrielBrascher** [comment]
@blueorangutan test

File: api/src/main/java/com/cloud/deploy/DeployDestination.java (hunk 1 above)
Comment (1m2c3t4) [reply api/src/main/java/com/cloud/deploy/DeployDestination.java]: added `this.` as suggested

**Comment by 1m2c3t4** [comment]
Thanks for the reviews @GabrielBrascher @DaanHoogland @rafaelweingartner

**Pull request merged**